### DISCUSSION TITLE
I seem to of lost a person!

### DISCUSSION THREAD

**Discussion by danmed** [photoprism/photoprism#2159] [Q&A] [answered]
I'm mid way through indexing all my photos and the scan stopped (i think i did something).. when i looked i noticed it had found a load of "people" which is great!

I was then looking through and it had grouped a few incorrectly.. so i was going through and retagging them to the right people.. but at some point something happened.. i ended up on a screen with just 1 person with an icon that kind of looked like a load of pictures stacked on top of each other...

now the original person with over 1000 images detected is gone.. i can't find a way to get them back group as a person.. they aren't hidden as far as i can tell ( i became a sponsor to find that out lol)... I'm concerned that these faces will never be matched again as i've done something stupid!

Is there anyone that can understand what i'm rambling about that might be able to help?

Edit : Ok i think i must of hit the "bad match" option on the entire person.. maybe? argh.. is it ever going to find the faces again and group them back together?

**Comment by danmed** [comment]
I'm also a bit confused about the unrecognised section and the all new people section.. if it's figured out there's a face.. why aren't they been grouped as a person?

**Reply by lastzero** [reply]
There's a RECOGNIZED and NEW, where do you see UNRECOGNIZED?

**Comment by danmed** [comment]
Apologies I mean recognised..

**Reply by lastzero** [reply]
It's the people you recognized by name.

**Comment by danmed** [comment]
Ok maybe I'm not explaining things... There is recognised, new and then at
the bottom of new there is "show all new faces" ... Why are they not
grouped like the others?

**Reply by lastzero** [reply]
https://docs.photoprism.app/user-guide/organize/people/

**Comment by danmed** [comment, marked as answer]
Yeah I've read that.. what I'm not sure on and it's related to my initial
question really is.. the facesi accidentally say as miss matched.. will
they, aling with those not yet grouped eventually be regrouped?

I'm still scanning so I assume that won't happen until the scan finishes?

**Reply by lastzero** [reply]
Please wait until indexing is done.

**Reply by danmed** [reply]
OK so.. indexing is done.. and there's a few issues..

1. all the photos i accidentally marked as mismatched are still not assigned to a person.. i can't find a way to bulk mark them either which means i'd have to do each one manually?
2. Some of the people are mixed up.. so i've gone in and marked 1 photo as mismatched.. i was assuming that having done that it might also pick up others that a re wrong and move them.. the logs seemed to back me up there : 

time="2022-03-17T10:02:06Z" level=info msg="faces: revised 151 matches after conflict"

However only that single photo i changed is now assigned to the correct person..

What is the advice on sorting this out? As far as i can tell nothing is going on right now.. no more facial recognition is being processed in the background and i have thousands of photos in "Show all new faces"... 

Is there some learning that will go on here? As i start to do things manually will it learn and start doing things itself?

**Reply by danmed** [reply]
I've run the below to start again with faces.. just waiting for something to happen now.. is there some kind of more verbose logging i can turn on so i know it's doing something?

docker-compose exec photoprism photoprism faces reset -f

**Reply by graciousgrey** [reply]
1. If you reject a face from a person it does not necessarily mean that this face can be automatically matched to another face cluster. Improvements for adding multiple faces to a person at once are planned (https://github.com/photoprism/photoprism/issues/1679) as well as many other things: https://github.com/photoprism/photoprism/issues?q=is%3Aissue+is%3Aopen+people+in%3Atitle

2. If you manually reject faces from a person or add faces to a person the face clusters are constantly updated in the background (PhotoPrism is learning). Also if you add new photos with faces on it this may lead to new clusters being formed. 
You saw this "learning"  when PhotoPrism removed additional faces from a person when you rejected a face. Just because PhotoPrism learned that some faces do not belong to a person it does not necessarily mean that it can match those faces to another person.

If you are not performing changes such as naming more faces or adding more photos there is nothing to learn.

Note that facial recognition is never perfect. In general it works best the more photos you have and with high quality thumbails, so make sure you are using at least default settings here (https://docs.photoprism.app/user-guide/settings/advanced/#images).

**Reply by graciousgrey** [reply]
You can turn on debug logs in settings

**Reply by danmed** [reply]
thank you, now that i've wiped all the faces.. do i just wait and eventually it'll start finding faces again or do i have to do a complete re-scan?

**Reply by graciousgrey** [reply]
After you have run reset you need to index faces again. Therefore you can use the faces index command https://docs.photoprism.app/getting-started/updates/#facial-recognition. Before doing manual changes wait until the face index has finished. You could also do a complete re-scan but this would take much longer.

**Reply by danmed** [reply]
thank you, i've been completely blind.. i've looked at that doc multiple times.. i think i was expecting the command to re-scan to be listed after the command to reset all the faces..

Right.. time to sit back and stop tampering with it.
